guppylang/tys/builtin.py

```diff
@@ -145,6 +175,10 @@ def bool_type() -> OpaqueType:
     return OpaqueType([], bool_type_def)
 
 
+def int_type() -> NumericType:
+    return NumericType(NumericType.Kind.Int)
```

Comment:
Should this not contain some reference to `int_type_def`?

Reply:
Is this used? I see a couple of places it'd work in this PR but it's just being inlined

Reply:
> Should this not contain some reference to int_type_def?

No, only the `int_type_def` has a reference to a `NumericType`. The inverse mapping from type to type def is implemeted here:
https://github.com/CQCL/guppylang/blob/cf8a5299e7b317de1f30f2c84c1b665bfba8c904/guppylang/checker/core.py#L82-L112

> Is this used? I see a couple of places it'd work in this PR but it's just being inlined

Good point, it's indeed unused. I'll remove it 👍